### PR TITLE
chore: upgrade rollup

### PR DESCRIPTION
**Pull request by g1eny0ung**
Signed-off-by: Yue Yang <g1enyy0ung@gmail.com>

**Overview**

This PR upgrade the rollup configuration, below are the changes:

- Upgrade rollup related deps
- Remove `namedExports` because the commonjs plugin can resolve named exports automatically now
- Replace `rollup-plugin-babel` to `@rollup/plugin-babel`. After my test, the bundling speed can be shortened by 2-3s

**Screenshots (if applicable)**

**Documentation**
- [ ] Updated Typescript types and/or component PropTypes
- [ ] Added / modified component docs
- [ ] Added / modified Storybook stories

### CODE REVIEW

**Comment by g1eny0ung** [comment]
@mshwery Hi is there any progress on this PR?

**Review by mshwery — COMMENTED** [review]
Hey @g1eny0ung this looks good to me! Could you try running `npx yarn-deduplicate && yarn install`? It looks like there were enough lockfile changes that there might be some overlap across deps.

@akleiner2 mind taking a look at this?

**Comment by g1eny0ung** [comment]
> Hey @g1eny0ung this looks good to me! Could you try running `npx yarn-deduplicate && yarn install`? It looks like there were enough lockfile changes that there might be some overlap across deps.

@mshwery Already done. 😆

**Comment by g1eny0ung** [comment]
@mshwery @akleiner2 The stalebot's comment made me notice that there are some conflicts in this PR. I already fix them. PTAL again.

**Comment by mshwery** [comment]
Looks like this is failing on size-limit checks. I think we can bump slightly.

**Comment by g1eny0ung** [comment]
@mshwery After resolving the conflict, I locally passed the size-limit check. Now CI also passed.

File: rollup.config.js [hunk 1]
```diff
@@ -1,31 +1,15 @@
...
-    'array',
-    'arrayOf',
-    'func',
-    'number',
-    'object',
-    'oneOf',
-    'oneOfType',
-    'string'
-  ]
-}
```
Comment (mshwery) [inline rollup.config.js:27]: Was this no longer needed in modern versions of the rollup commonjs plugin?

**Comment by mshwery** [comment]
I compared the output (unminified) and it's significantly different in size than the previous output. Seeing an initial output of 1.6mb → 1.8mb

**Comment by mshwery** [comment]
I would be ok merging this only because it's a tertiary output format (UMD) that isn't used by most folks, but would like more confidence in that build output size change.

File: rollup.config.js (hunk 1 above)
Comment (g1eny0ung) [reply rollup.config.js:27]: You can refer to these two PRs for more detail:

https://github.com/rollup/rollup/pull/3295
https://github.com/rollup/plugins/pull/149

In short, with `syntheticNamedExports`, rollup will automatically extract **named exports** from commonjs `exports`.

**Comment by g1eny0ung** [comment]
> I compared the output (unminified) and it's significantly different in size than the previous output. Seeing an initial output of 1.6mb → 1.8mb

I tested it locally and it doesn't seem to change anything:

image

**Review by mshwery — APPROVED** [review]
Just tested again and you are correct! Not sure what was up with the last time I tested your PR.

**Pull request merged**